ee/tabby-ui/app/files/components/action-bar-widget/action-bar-widget.tsx

```diff
@@ -0,0 +1,74 @@
+import Image from 'next/image'
+import tabbyLogo from '@/assets/tabby.png'
+
+import { CodeBrowserQuickAction, emitter } from '@/lib/event-emitter'
+import { cn } from '@/lib/utils'
+import { Button } from '@/components/ui/button'
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger
+} from '@/components/ui/dropdown-menu'
+import { IconChevronUpDown } from '@/components/ui/icons'
+
+interface ActionBarWidgetProps extends React.HTMLAttributes<HTMLDivElement> {}
+
+export const ActionBarWidget: React.FC<ActionBarWidgetProps> = ({
+  className,
+  ...props
+}) => {
+  const handleAction = (action: CodeBrowserQuickAction) => {
+    emitter.emit('code_browser_quick_action', action)
```

Comment:
The reasons for using an event emitter:
 1 .The widget component cannot access the app's context, so a global eventEmitter is used to trigger actions.
 2. To prevent widget flicker, the widget itself is cached and cannot obtain the latest selected text.